projects/default.nix

```
@@ -0,0 +1,42 @@
+{
+  lib,
+  pkgs ? {},
+  ...
+} @ args: let
+  baseDirectory = ./.;
+
+  inherit
+    (builtins)
+    pathExists
+    readDir
+    trace
+    ;
+
+  inherit
+    (lib.attrsets)
+    mapAttrs
+    concatMapAttrs
+    recursiveUpdate
+    ;
+
+  names = name: type:
+    if type != "directory"
+    then assert name == "README.md" || name == "default.nix"; {}
+    else {${name} = baseDirectory + "/${name}";};
+
+  allProjectDirectories = concatMapAttrs names (readDir baseDirectory);
+
+  projectDirectories = lib.filterAttrs (_: directory:
+    if pathExists (directory + "/project.nix")
+    then true
+    else trace "No project.nix found in ${directory}, skipping." false)
+  allProjectDirectories;
+in
+  mapAttrs (
+    _: directory: let
+      imported = import (directory + "/project.nix") args;
```

(not tested)

```suggestion
  mapAttrs (
    _: directory: let
      imported = import directory args;
```

Can't we just essentially say that each project dir must have a `default.nix` and everything will fall apart if any of them doesn't?

---

Yes, though "falling apart" is not very nice for developing. For example we have the case of LiberaForms-E2EE which currently is unmaintained (#40). In such cases one might want to keep the directory and somehow have an option to skip it.

Of course, there's the alternatives of just deleting that stuff, or marking it as broken.

---

Let's just mark as broken.

---

I added handling for `(import ./projects/<name>).broken` in `./projects/default.nix`.